api/tests/python/models/test_request.py

```diff
@@ -107,6 +112,36 @@ def test_has_consumed_name():
 
     assert nr.has_consumed_name is True
 
+@pytest.mark.parametrize(['testName','compName', 'compState', 'nrNum','nrEntity','assertValue'], [
+    ('Sole Prop Approved Company', 'TEST1', 'APPROVED','NR 0000001', 'FR', False),
+    ('Gen Partner Approved Company','TEST2', 'APPROVED','NR 0000002', 'GP', False),
+    ('CORPORATION Approved Company','TEST3', 'APPROVED','NR 0000003', 'CR', True),
+    ('Sole Prop Conditional Company','TEST4', 'CONDITION','NR 0000004', 'FR', False),
+    ('Gen Partner Conditional Company','TEST5', 'CONDITION','NR 0000005', 'GP', False),
+    ('CORPORATION Conditional Company','TEST6', 'CONDITION','NR 0000006', 'CR', True),
+])
+def test_no_cloud_event_sent_for_sp_gp_nrs(testName, compName, compState, nrNum, nrEntity, assertValue):
+    """func description"""
+    from namex.models import Name, Request as RequestDAO, State
+    
+    with patch.dict(current_app.config, {'DISABLE_NAMEREQUEST_NATS_UPDATES': 0}):
+        with patch.object(queue_util, 'send_name_request_state_msg', return_value=True) as mock_queue_util_send_msg_func:
+            name = Name()
+            name.choice = 1
+            name.name = compName
+            name.state = compState
+
+            nr = RequestDAO()
+            nr.nrNum= nrNum
+            if compState == 'APPROVED':
```

Comment:
i think it's ok to just use the parameter value directly without doing the if/else for testing purposes.  simplify things.

Reply:
i had trouble passing State.approved and State.conditional that's why i have the if condition

Reply:
those are just strings so you should be able to just pass them in as is.  you might have had troubles assigning `CONDITION` to `nr.stateCd`.  `CONDITIONAL` is a valid value.  `CONDITION` isn't.